### PR TITLE
compare: output improvements

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
Fixes: #650

A few concerns have been voiced about the west compare output:

   1. its signal:noise ratio is too low (too many lines per
      project without enough useful information per line)
   2. the end of the status output from one project is hard to
      separate from the status output for the next
   3. it isn't obvious enough whether HEAD == manifest-rev or not

To try to resolve 2. without making 1. any worse, explicitly print
HEAD and manifest-rev if they differ, without adding any extra lines,
by consolidating some of the output into the relevant banner line.

To solve 3., indent the status output to the same level as the
'status:' text in the banner.

### CODE REVIEW

**Comment by mbolivar-nordic** [comment]
We could go even further with something like this:

image

LMK what y'all want me to do

**Review by marc-hb — APPROVED** [review]
I think there are probably ways to make it even better but this looks like good visual progress to me and the internal refactoring looks like a good starting point for further improvements like the suggestion I just made in
- #650

> We could go even further with something like this:

No because it would make the `manifest-rev:` prefix and SHA1 start in a variable column.

**Comment by mbolivar-nordic** [comment]
> Was this change intended?

It was not; now fixed.

**Comment by mbolivar-nordic** [comment]
image

**Comment by mbolivar-nordic** [comment]
> * [Still difficult to tell for which repos HEAD == manifest-rev in `west compare` #650](https://github.com/zephyrproject-rtos/west/issues/650)

I'm starting to feel like the extra color and full `git log --oneline` output is noisier than it should be.

I could do something like this in this PR:

```
--- manifest-rev: cafef00d drivers: add all the bugs
            HEAD: deadbeef drivers: fix all the bugs
--- status:
    [...indented, colorized text goes here...]
```

Where the manifest-rev and HEAD lines are shown only if the two revisions are different SHAs, and no color is present in any case.

**Comment by aescolar** [comment]
> It was not; now fixed.

This looks very fine to me

> Where the manifest-rev and HEAD lines are shown only if the two revisions are different SHAs

If very fine with this proposal.

**Comment by mbolivar-nordic** [comment]
> Im very fine with this proposal.

OK, now done; looks like this:

image

Where the first banner is not printed at all if HEAD and manifest-rev are the same commit.

**Comment by mbolivar-nordic** [comment]
@marc-hb since this has changed substantially since your first approval, I will give you some time to look again before merging

File: src/west/app/project.py
Comment (marc-hb) [inline src/west/app/project.py:740]: This comment was the rationale for running `git log` ONCE. Now that you are running `git log` twice it feels a bit out of place. It's still useful information though, can you move it below next to `git status`?

File: src/west/app/project.py
Comment (marc-hb) [inline src/west/app/project.py:735]:
```suggestion
        # We force git log not to print in color so west colored separators stand out
        # more.
```
git status is still colored by default.

File: src/west/app/project.py
Comment (marc-hb) [inline src/west/app/project.py:748]: I would drop this because this is the `west compare` command and this is not comparing anything. Also, it can be misleading when scanning the output quickly for repos which are not on the manifest.

**Review by marc-hb — COMMENTED** [review]
I don't think it helps with #648 much, especially not with uncolored `west compare | less` but it does not make it worse either.

On the other hand this really helps with #650, thanks!

Small changes suggested below.

**Comment by mbolivar-nordic** [comment]
> I don't think it helps with #648 much, especially not with uncolored `west compare | less` but it does not make it worse either.

I am surprised to hear you say that. I think the indentation makes the separation between status output and the green lines quite easy to read, personally. Are you saying an extra newline reads significantly more clearly to you than an indented block?

**Comment by mbolivar-nordic** [comment]
> and the green lines quite easy to read, personally.

... and indentation is of course preserved across pipelines that remove color

**Comment by marc-hb** [comment]
> I think the indentation makes the separation between status output and the green lines quite easy to read, personally. Are you saying an extra newline reads significantly more clearly to you than an indented block?

I would take both but after taking a closer look at the before/after difference you're right: the indentation does help quite a lot. Apologies.

**Comment by mbolivar-nordic** [comment]
To make it easier to review, here's a screenshot of how it looks as of now:

image

**Review by marc-hb — APPROVED** [review]
Still two confusing and/or misplaced comments and some strange color behavior (what's new?) but that shouldn't block this massive improvement, thanks!

File: src/west/app/project.py
Comment (mbolivar-nordic) [reply src/west/app/project.py:756]: > Any idea why `git -c status.color=always status | less -R` shows colors but `west compare | less -R` does not with this PR?

west won't do color if stdout is not a tty

Comment (marc-hb) [reply src/west/app/project.py:753]: Answering myself after a chat with @mbolivar-nordic : `colorama` has no "always" concept and always strips colors when connected to a terminal. Commenting out `colorama.init()` "fixes" `west compare  | less -R`

**Pull request merged**
